pkg/kubelet/rocket/rocket.go

```
@@ -0,0 +1,306 @@
+		return nil, nil
+	}
+	for _, line := range strings.Split(strings.TrimSpace(string(output)), "\n") {
+		tuples := strings.Split(strings.TrimSpace(line), "\t")
+		if len(tuples) != 3 { // HARDCODE, non state line.
+			continue
+		}
+		status[tuples[0]] = &PodStatus{
+			State: tuples[2],
+		}
+	}
+	return status, nil
+}
+
+func (r *RocketRuntime) unitFileToPod(fname string) (*api.Pod, error) {
+	f, err := os.Open(fname)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	var pod api.Pod
+	opts, err := unit.Deserialize(f)
+	if err != nil {
+		return nil, err
+	}
+
+	var rktID string
+	for _, opt := range opts {
+		if opt.Section != "X-K8S" {
+			continue
+		}
+		if opt.Name == "RocketID" {
+			rktID = opt.Value
+		}
+		if opt.Name == "POD" {
+			// NOTE: In fact we unmarshal from a serialized
+			// api.BoundPod type here.
+			err = json.Unmarshal([]byte(opt.Value), &pod)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// TODO(yifan): Cache the list result
+	podState := "unknown"
+	pods, err := r.getPodsStatus()
+	if err == nil {
+		if _, found := pods[rktID]; !found {
+			return nil, fmt.Errorf("pod not running from rkt list")
+		}
+		podState = pods[rktID].State
+	}
+
+	fmt.Println("pod state", rktID, podState)
+
+	pod.Status.Info = make(map[string]api.ContainerStatus)
+	for _, container := range pod.Spec.Containers {
+		switch podState {
+		case "running":
+			pod.Status.Info[container.Name] = api.ContainerStatus{
+				State: api.ContainerState{
+					Running: &api.ContainerStateRunning{},
+				},
+			}
+		case "embryo", "preparing", "prepared":
+			pod.Status.Info[container.Name] = api.ContainerStatus{
+				State: api.ContainerState{
+					Waiting: &api.ContainerStateWaiting{},
+				},
+			}
+		case "exited", "deleting", "gone":
+			pod.Status.Info[container.Name] = api.ContainerStatus{
+				State: api.ContainerState{
+					Termination: &api.ContainerStateTerminated{},
+				},
+			}
+		default:
+			panic("Unexpected state")
```

Again with the panic. Why panic rather than return an error?